x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/Autoscaling.java

```diff
@@ -69,7 +71,7 @@ public Autoscaling(final Settings settings) {
      */
     @Override
     public List<Setting<?>> getSettings() {
-        if (isSnapshot() || AUTOSCALING_FEATURE_FLAG_REGISTERED) {
+        if (isSnapshot() || (AUTOSCALING_FEATURE_FLAG_REGISTERED != null && AUTOSCALING_FEATURE_FLAG_REGISTERED)) {
```

Comment:
With my previous suggestion, this simplifies to the following:

```suggestion
        if (AUTOSCALING_FEATURE_FLAG_REGISTERED) {
```